experimenter/experimenter/nimbus_ui/forms.py

```diff
@@ -1541,3 +1543,55 @@ def __init__(self, *args, **kwargs):
         }
         self.fields["application"].widget.attrs.update(htmx_attrs)
         self.fields["feature_configs"].widget.attrs.update(htmx_attrs)
+
+
+class TagForm(forms.ModelForm):
+    name = forms.CharField(
+        required=True,
+        max_length=100,
+        label="Tag Name",
+        widget=forms.TextInput(attrs={"class": "form-control"}),
+    )
+    color = forms.CharField(
+        required=True,
+        max_length=7,
+        label="Color",
+        widget=forms.TextInput(
+            attrs={"type": "color", "class": "form-control form-control-color"}
+        ),
+    )
+
+    class Meta:
+        model = Tag
+        fields = ["name", "color"]
+
+
+class TagBaseFormSet(BaseModelFormSet):
+    def clean(self):
+        if any(self.errors):
+            return
+
+        names = []
+        for form in self.forms:
+            name = form.cleaned_data.get("name")
+            if name:
+                names.append(name.lower())
+
+        if len(names) != len(set(names)):
+            raise forms.ValidationError(NimbusUIConstants.ERROR_TAG_DUPLICATE_NAME)
+
+    def create_tag(self):
+        # Create a new tag with a unique name and random color
+        base_name = "Tag"
+        existing_names = set(Tag.objects.values_list("name", flat=True))
+        idx = 1
+        while f"{base_name} {idx}" in existing_names:
+            idx += 1
```

Comment:
Ya you can do it this way, or you can use a set method:

https://github.com/mozilla/experimenter/blob/main/experimenter/experimenter/nimbus_ui/forms.py#L780-L783